android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactoryTest.java

```diff
@@ -223,6 +234,118 @@ public void testRepeatingGroupDefaultCount() throws Exception {
         Assert.assertEquals("0", repeatingGroupCountObj.getString(JsonFormConstants.VALUE));
     }
 
+    @Test
+    public void testAddOnDoneActionShouldCreateRepeatingGroups() throws InterruptedException, JSONException {
```

Comment:
What is the main difference between this and https://github.com/OpenSRP/opensrp-client-native-form/pull/562/files#diff-11c7b636560d97f26105e24c3f41db27f48a562c3852620ab9400402df5481c5R287? There seems to be a lot of duplicated coded between the two.

Reply:
So when someone creates a repeating group this is called `factory.addOnDoneAction(textView, imageButton, widgetArgs);`  e.g creating 2 repeating groups ... then if say someone changes the number to 1 the same line `factory.addOnDoneAction(textView, imageButton, widgetArgs);` will be called again to remove the excess fields ... thats why in the test i create 2 groups initially then change the number in the reference edit text to 1 then invoke the same method.

Reply:
So instead of mocking the views to be updated i called `factory.addOnDoneAction(textView, imageButton, widgetArgs);` twice

Reply:
OK, could we move the common code to a private method? And rename the tests to describe what is happening? That it's testing deletion of fields?